### PR TITLE
Fixes #2670 - Makes sure we are passing a JSON object for processing details.

### LINKED ISSUE #2670: Take into account that 'null' as a value for details is  a valid JSON
@miketaylr Current staging is crashing on reporting issue.

```
[2018-10-24 06:14:13,421] INFO in views: 210.194.43.135 http://14.3.0.example.com/
Traceback (most recent call last):
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 2309, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 2295, in wsgi_app
    response = self.handle_exception(e)
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 1741, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/webcompat/staging.webcompat.com/env/local/lib/python2.7/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "./webcompat/views.py", line 242, in create_issue
    json_response = report_issue(form, proxy=True)
  File "./webcompat/issues.py", line 31, in report_issue
    data=json.dumps(build_formdata(form)))
  File "./webcompat/form.py", line 357, in build_formdata
    body += build_details(details)
  File "./webcompat/form.py", line 161, in build_details
    console_logs = content.pop('consoleLog', None)
AttributeError: 'NoneType' object has no attribute 'pop'
```

This is what is breaking it.
https://github.com/webcompat/webcompat.com/blob/1f5947956f783cfd2dbcb8de2e01fb6c5443a4eb/webcompat/form.py#L157-L165

This was written for solving issue #2653 with the request #2658

So basically we get an issue when the `content` is `None`

`json.loads(details)` might return None. We need to find out why.

One way to have a more robust way of doing things here.

### PR DESCRIPTION
**Pull request by karlcow**
This PR fixes issue #2670

## Proposed PR background

This fixes a couple of issues, but mainly it makes sure that if details is not a JSON object we catch it and return a string. For this it adds a new method in helpers to check if the JSON is an object.
It also fixes a couple of minor syntactic issues.

### CODE REVIEW

File: tests/unit/test_helpers.py [hunk 1]
```diff
@@ -48,11 +50,15 @@
 
 
 class TestHelpers(unittest.TestCase):
+    """Clas for testing helpers."""
```
Comment (miketaylr) [inline tests/unit/test_helpers.py]:
```suggestion
    """Class for testing helpers."""
```

File: tests/unit/test_helpers.py [hunk 2]
```diff
@@ -301,5 +307,14 @@ def test_prepare_form_post(self):
                 method='POST'):
             self.assertEqual(prepare_form(flask.request), json_data)
 
+    def test_json_array(self):
+        """Check if we return the right type of each JSON."""
+        # A simple JSON array
```
Comment (miketaylr) [inline tests/unit/test_helpers.py]: Do we mean object instead of array?

File: webcompat/form.py [hunk 3]
```diff
@@ -129,7 +130,7 @@ def get_details(details):
         rv = ''.join(['<li>{k}: {v}</li>'.format(k=k, v=get_str_value(v))
                       for k, v in details.items()])
     except AttributeError:
-        return content
+        return '<li>{content}</li>'.format(content=content)
```
Comment (miketaylr) [inline webcompat/form.py]: This seems like such an obviously good idea... wish I would have thought of it. :)

File: webcompat/helpers.py [hunk 4]
```diff
@@ -626,3 +627,8 @@ def prepare_form(form_request):
         json_data = form_request.get_json()
         form_data.update(json_data)
     return form_data
+
+
+def is_json_array(json_data):
```
Comment (miketaylr) [inline webcompat/helpers.py]: I think we want to call this `is_json_object` for clarity. When I think an array, I think of something that resembles a list: `[1,2,3]`. 

Or, maybe I'm not considering something else?

File: tests/unit/test_helpers.py (hunk 2 above)
Comment (karlcow) [reply tests/unit/test_helpers.py]: And I meant an `object`. doh! And here goes all my theatrical makeup over my face. Thanks.

File: webcompat/helpers.py (hunk 4 above)
Comment (karlcow) [reply webcompat/helpers.py]: nope you are right. :) my bad.

File: tests/unit/test_helpers.py (hunk 1 above)
Comment (karlcow) [reply tests/unit/test_helpers.py]: done

File: webcompat/form.py (hunk 3 above)
Comment (karlcow) [reply webcompat/form.py]: ❤️

**Comment by karlcow** [comment]
@miketaylr taken into account your comments. Thanks a lot!
Rebased and pushed again.

```
→ nosetests
...........................................................................................................
----------------------------------------------------------------------
Ran 107 tests in 1.979s

OK
```

**Comment by karlcow** [comment]
Rha one more rebase to do. Hold on.

**Comment by karlcow** [comment]
ok I just adjusted the commit messages too.

**Comment by karlcow** [comment]
@miketaylr CircleCI is happy

**Pull request merged**